pkg/build/graph/edges.go

```diff
@@ -55,24 +57,37 @@ func AddInputOutputEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildCon
 		g.AddEdge(in, node, BuildInputEdgeKind)
 	}
 
+	sourceImages := []*kapi.ObjectReference{}
 	from := buildutil.GetImageStreamForStrategy(node.BuildConfig.Spec.Strategy)
 	if from != nil {
-		switch from.Kind {
+		sourceImages = append(sourceImages, from)
```

Comment:
hm.  that is not correct behavior because build-chain is supposed to report "what will be rebuilt if this image changes (which requires an ICT exist)" not "what is dependent on this image/what *should* be rebuilt if this image changes".

at least that's what it used to report.  if we've changed the behavior we need to make sure it's doc'd and consider re-introducing the old behavior as an option as well.  @kargakis @deads2k